spring-cloud-config-client/src/main/java/org/springframework/cloud/config/client/ConfigServicePropertySourceLocator.java

```diff
@@ -138,6 +155,15 @@ public org.springframework.core.env.PropertySource<?> locate(org.springframework
 
 	}
 
+	// to check if different profile is activated within the default profile
+	private boolean checkIfProfileIsActivatedInDefault(String[] profiles, Map<String, Object> map) {
+		List<String> profilesList = Arrays.asList(profiles);
+		return (profilesList.size() == 1 && profilesList.get(0).equalsIgnoreCase("default")
```

Comment:
@spencergibb do you recall why this would only need to be done for the default profile?  What if the default profile contains `spring.profiles.active` and then subsequent configuration properties which get activated based on that profile also contain `spring.profiles.activate`?  In that case we would not also activate those profiles.